### PR TITLE
fix: hint overlap

### LINKED ISSUE #6230: [Bug]-[348]:Slash command dropdown overflow
## Description The slash comment should overflow below the field out of the property pane and should not hide the field inside the property pane.

[What happened]

 The dropdown overlaps on the property pane in the below use cases:

- [ ] Slash commands menu opens up on top when there is no enough space at the bottom.
- [ ] when the code is written beyond the 2nd line the dropdown opens in the top
- [ ] When the input box is very wide the dropdown opens up in the random place

Widgets | Property pane fields where the overlap happens
-- | --
Table widget |  
  | Default Search Text
  | Default Selected Row
  | Actions
  | onRowSelected
  | onPageChange
  | onPageSizeChange
  | onSearchTextChanged
Button |  
  | Visible
  | Google Recaptcha Key
  | Actions
chart |  
  | Series Title
  | Series Data
  | Axis
  | Actions
checkbox |  
  | Visible
  | Disabled
  | Actions
Container |  
  | Background Color
  | Visible
Datepicker |  
  | Required
  | Visible
  | Disabled
  | Min Date
  | Max Date
  | Actions
Divider |  
  | Thickness (px)
FilePicker |  
  | Allowed File Types
  | Required
  | Visible
  | Disable
  | Actions
Iframe |  
  | onMessageReceived
  | Border Opacity (%)
  | Border Width (px)
  | nURLChanged(The dropdown is not displayed correct)
  | onMessageReceived(The dropdown is not displayed correct)
Image |  
  | Max Zoom Level
  | Object Fit
  | Actions
Input |  
  | Regex
  | Error Message
  | Required
  | Visible
  | Disable
  | Reset on submit
  | Actions
List |  
  | Item Spacing (px)
  | Visible
  | Actions
Map |  
  | Visible
  | Actions
menu |  
  | Visible
  | Compact
  | Actions
radio |  
  | Options
  | Default Selected Value
  | Required
  | Visible
  | Disabled
  | Actions
rating |  
  | Tooltips
  | Allow half stars
  | Visible
  | Disabled
  | Actions
Rich text editor |  
  | Default text
  | Required
  | Visible
  | Disabled
  | Actions
select dropdown |  
  | Options
  | Default Option
  | Filterable
  | Required
  | Visible
  | Disabled
  | Actions
switch |  
  | Default Selected
  | Visible
  | Disabled
  | Actions
tab |  
  | Visible
  | Actions
Text |  
  | Text Color
  | Font Style
  | Text Align
Video |  
  | Visible
  | Actions

### Important Details

- Version: [Cloud ]
- OS: [MacOSX]
- Browser chrome
- Environment [production]

### PR DESCRIPTION
**Pull request by Rishabh-Rathod**
## Description

Removed the `margin-top: ${(props) => props.theme.spaces[3]}px;` style as it was only valid for the case of autocomplete suggestion opening below the code.
- When a popup was shown above in those cases it overlapped the code.

Fixes #6230

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

Manual test cases
- [ ] When slash commands popup opens up above the property assert that it doesn’t overlap the code.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by Rishabh-Rathod — COMMENTED** [review]
self-review is done. LGTM.

File: app/client/src/utils/autocomplete/TernServer.test.ts [hunk 1]
```diff
@@ -151,7 +151,8 @@ describe("Tern server", () => {
             end: { ch: 6, line: 0 },
           },
         },
-        expectedOutput: { ch: 2, line: 0 },
+        // we will receive end position in `from`
+        expectedOutput: { ch: 6, line: 0 },
```
Comment (Rishabh-Rathod) [inline app/client/src/utils/autocomplete/TernServer.test.ts:155]: completion.text.length = 4
cursor start = 2
from ( end ) = 4 + 2

**Review by Rishabh-Rathod — COMMENTED** [review]
Fixed jest test correctly. Code LGTM

**Review by ohansFavour — APPROVED** [review]

**Review by ohansFavour — CHANGES_REQUESTED** [review]
Get design sign-off first

**Comment by Rishabh-Rathod** [comment]
Closing PR. 

Internal slack discussion:- https://theappsmith.slack.com/archives/C02K0SZQ7V3/p1664165909501019?thread_ts=1663320235.839129&cid=C02K0SZQ7V3

**Pull request closed without merging**